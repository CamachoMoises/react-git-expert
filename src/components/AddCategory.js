import React, { useState } from 'react';

export const AddCategory = () => {
    const [inputValue, setInputValue] = useState('New category')

    const handleInputChange=(e)=>{
        setInputValue(e.target.value)
    }
    const handleSubmit= (e)=>{
        e.preventDefault()
        console.log('submitted');
    }   
	return (
		<form onSubmit={handleSubmit}>
			<h2> Add Category {inputValue} </h2> 
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />

		</form>
	);
};
