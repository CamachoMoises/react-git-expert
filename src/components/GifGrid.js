import React, { useEffect, useState } from 'react';

export const GifGrid = ({ category }) => {
	const [images, setImages] = useState([]);
	useEffect(() => {
		getGif();
	}, []);

	const getGif = async () => {
		const url = 'https://api.giphy.com/v1/gifs/search?q=Speed+Racer&limit=10&api_key=wnzsNoA0er1NfL4znwh2pMAqRWa5BTi2';
		const resp = await fetch(url);
		const { data } = await resp.json();
		const gifs = data.map((gif) => {
			return {
				id: gif.id,
				title: gif.title,
				url: gif.images?.downsized_small.mp4,
			};
		});
		console.log(gifs);
		setImages(gifs);
	};

	return (
		<div>
			<h3> {category} </h3>
			<ol>
				{images.map(({id, title}) => (
					<li key={id}> {title}</li>
				))}
			</ol>
		</div>
	);
};
