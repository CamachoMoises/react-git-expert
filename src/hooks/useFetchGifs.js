import { useEffect, useState } from "react"
import getGif from "../helpers/getGifs"


const useFetchGifs= (category)=>{
    const [state, setState] = useState({
        data:[],
        loading:true,
    })
    useEffect(() => {
        getGif(category).then(img => {
            setTimeout(() => {
                setState({
                    data:img,
                    loading:false,
                })
            }, 3000);
        })
    }, [category])

    return state // {data:[], loading:true}

}

export default useFetchGifs