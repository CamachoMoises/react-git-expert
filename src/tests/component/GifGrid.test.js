import React from 'react'
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test in GifGrid component', () => {
	const category = 'test';
	beforeEach(() => {});
    
	test('should show the component snapshot', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            load: true
        })
        let wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});
	test('should show the items when is charge images from useFetchGif', () => {
        const gifs=[{
            id:'ABC',
            url:'http://example.com',
            title:'test title'
        },{
            id:'DEF',
            url:'http://example.com',
            title:'test titless'
        }
    ]
        useFetchGifs.mockReturnValue({
            data:gifs,
            load: false
        })
        
        
        let wrapper = shallow(<GifGrid category={category} />);
        
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
		expect(wrapper).toMatchSnapshot();
    });
});
