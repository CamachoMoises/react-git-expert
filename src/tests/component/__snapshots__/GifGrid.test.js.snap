// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGrid component should show the component snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    test
     
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test in GifGrid component should show the items when is charge images from useFetchGif 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    test
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="test title"
      url="http://example.com"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="test titless"
      url="http://example.com"
    />
  </div>
</Fragment>
`;
